Adds emag effects to the mini-nuke and meteor toys

TheVekter opened the pull request:
## About The Pull Request

Adds some fun little emag effects to some of the toy items. The mini-nuke toy will actually explode when the countdown finishes while the mini meteor will blow up and damage nearby walls.

## Why It's Good For The Game

Because they are fun little ways to add some flavor to existing items.

Changelog
🆑
add: Added emag effects to the mini-nuke and meteor toys.
/🆑

TheVekter committed:
Adds emag effects to the mini-nuke and meteor toys

TheVekter commented:
Maintainer permission under antag freeze:
![image](https://user-images.githubusercontent.com/7019927/70220716-9e3bff80-170c-11ea-8b57-52a418781e76.png)

Tlaltecuhtli commented:
does the explosion on the mini meteor break clothing / remove limbs / stun for 20s?

TheVekter commented:
> does the explosion on the mini meteor break clothing / remove limbs / stun for 20s?

It should _not_ but I will test again to be sure.

TheVekter committed:
Reduces range of mini-meteor explosion

TheVekter commented:
Okay, so it does break clothing and knock limbs off, but only on the person you throw it at. I'm removing the bouncing behavior to keep anything really fucky from happening.

TheVekter committed:
Normal meteor toy doesn't break after one throw, emagged one does.

ATH1909 commented on code/game/objects/items/toys.dm:
```diff
@@ -1032,20 +1032,39 @@
 		icon_state = "nuketoycool"
 		sleep(cooldown - world.time)
 		icon_state = "nuketoyidle"
+	else if (obj_flags & EMAGGED)
```
didn't you already have a check for obj_flags & EMAGGED earlier?

ATH1909 commented on code/game/objects/items/toys.dm:
```diff
@@ -1032,20 +1032,39 @@
 
 /obj/item/toy/nuke/attack_self(mob/user)
-	if (cooldown < world.time)
+	if (obj_flags & EMAGGED)
```
this check happens before the cooldown check, so you can spam-use an emagged toy nuke

TheVekter replied:
I can re-add the check, but because the explosion deletes the toy nuke, this isn't relevant.

ATH1909 commented on code/game/objects/items/toys.dm:
```diff
@@ -1057,20 +1076,32 @@
+/obj/item/toy/minimeteor/emag_act(mob/user)
+	if (obj_flags & EMAGGED)
+		return
+	to_chat(user, "<span class = 'notice'> You short-circuit whatever electronics exist inside \the [src]. If there even are any.</span>")
```
sentence fragment bad

ATH1909 commented:
> Okay, so it does break clothing and knock limbs off, but only on the person you throw it at. I'm removing the bouncing behavior to keep anything really fucky from happening.

Aw, but the bouncing behavior is cool and might lead to someone blowing themselves up with their own mini-meteor!

Timonkeyn commented:
Make the toy emag emagable making it a real emag

Ziiro commented:
>Make the toy emag emagable making it a real emag

Better:
Emagging a toy emag makes it look like a real emag, but it's actually a bomb that goes off when it's used.

necromanceranne commented:
It's hard to get these consistently so this just seems cool and fun. I know this from having some events that involved collecting antag tokens. Just getting one with any amount of reliability is quite hard.

TheVekter commented:
I fixed all of these before leaving for work, I just need to test and make sure I didn't break anything before pushing it.



> It's hard to get these consistently so this just seems cool and fun. I know this from having some events that involved collecting antag tokens. Just getting one with any amount of reliability is quite hard.

This is a big reason why I don't mind the damage from the mini meteor being a little overtuned. You only get one throw, it's pretty hard to get, and it seems like a fun little bonus if you find one or win one. The other intention behind these is letting them act like a trap. What do you do if you just find a toy nuke lying around somewhere? You pick it up and use it. Now you're deep in red and less at least one limb.

TheVekter committed:
90% of the time I have no idea what the fuck I'm talking about

TheVekter commented:
Bunch of small changes in this commit. I reduced the cooldown on the nuke toy because three minutes was a little goofy. I implemented @ATH1909's suggestions and refactored the code a bit as well. You were able to spam the sound before, but it wouldn't explode multiple times, just gets rid of an annoyance.

A clever touch (at least IMO) was making the icon state not change on the emagged bomb, giving you a moment to realize it's not normal and yeet it elsewhere.

ATH1909 commented:
> A clever touch (at least IMO) was making the icon state not change on the emagged bomb, giving you a moment to realize it's not normal and yeet it elsewhere.

But... what logical reason would the icon have to NOT change if the item is emagged?

TheVekter commented:
> > A clever touch (at least IMO) was making the icon state not change on the emagged bomb, giving you a moment to realize it's not normal and yeet it elsewhere.
> 
> But... what logical reason would the icon have to NOT change if the item is emagged?

Because the emag shorted out the LEDs on it. It always shows the version without any lights.

ExcessiveUseOfCobblestone commented:
are they telling you it can't just be an emag effect? I don't understand how a pure emag effect isn't antag-exclusive

TheVekter commented:
Nah, they mean as long as I'm not adding an item that's antag exclusive.

optimumtact commented on code/game/objects/items/toys.dm:
```diff
@@ -1032,45 +1032,73 @@
+	if (obj_flags & EMAGGED && cooldown < world.time)
+		cooldown = world.time + 600
+		user.visible_message("<span class='hear'>You hear the click of a button.</span>", "<span class='notice'>You activate [src], it plays a loud noise!</span>")
+		sleep(5)
```
you have to check for item qdeleted afterevery time you sleep, how can people not know this by now

TheVekter replied:
Because I left for like a year and this is the first bit of code I've used with a sleep in it. What do I need to add to un-fuck it?

ExcessiveUseOfCobblestone commented on code/game/objects/items/toys.dm:
```diff
@@ -1032,45 +1032,81 @@
 /obj/item/toy/nuke/attack_self(mob/user)
-	if (cooldown < world.time)
-		cooldown = world.time + 1800 //3 minutes
+	var/timeleft = (cooldown - world.time)
```
timeleft is only used at the else so move it there

TheVekter replied:
Fixed.

ExcessiveUseOfCobblestone commented on code/game/objects/items/toys.dm:
```diff
@@ -1032,45 +1032,81 @@
+	if (obj_flags & EMAGGED)
+		return
+	to_chat(user, "<span class = 'notice'> You short-circuit \the [src].</span>")
+	desc = "A plastic model of a Nuclear Fission Explosive."
```
what's different between this desc and original

TheVekter replied:
Meant to remove this.

ExcessiveUseOfCobblestone commented on code/game/objects/items/toys.dm:
```diff
@@ -1032,45 +1032,81 @@
 		for(var/mob/M in urange(10, src))
 			if(!M.stat && !isAI(M))
 				shake_camera(M, 3, 1)
-		qdel(src)
```
did you mean to fully remove this? this means it won't delete whether it's emagged or not

TheVekter replied:
I did. The original idea I had for the toy was that you'd just use it once and it'd be gone, but I decided that's kind of... not fun. So screw it, no more qdel after being thrown once. If it proves to be annoying or a problem we can always revert it.

TheVekter replied:
Yeah. The original idea I had was that it would only be used once before it breaks, but I felt that was a lot less fun that way. I don't think it'll get abused and, if it does, we can just revert this.

TheVekter committed:
Refactor
